## Input Preparation
- input folder preparation: https://hubmapconsortium.github.io/ingest-validation-tools/rnaseq/current/

## config parameters
- `keep_all_barcode` : default depends on assay type. For 10x_v2, keep_all_barcodes is false by default.
- `assay` : passed by cli (10x_v2)
- `threads` : passed by cli
- `fastq_dir` : passed by cli

## Workflow descriptions
- your best bet is probably looking at a HuBMAP page: https://portal.hubmapconsortium.org/browse/dataset/f47349f1e0fea69263a8d68c7dac65c5
- There are a few descriptions in there, Matt might have a better description of the files.
- Files under hubmap-ui are the files that are generated for the HuBMAP front-end (vitessce).The zarr files are generated by this workflow: https://github.com/hubmapconsortium/portal-containers/tree/master/containers/anndata-to-uiThe other files are generated by this workflow: https://github.com/hubmapconsortium/portal-containers/tree/master/containers/h5ad-to-arrow
- The CWL for the salmon pipeline has some description of outputs: https://github.com/hubmapconsortium/salmon-rnaseq/blob/main/pipeline.cwl

## The list of output files generated:

### Data products (essential files of interest):
total 1266780
drwxr-xr-x. 5 tangmuna hubmap      4096 Mar 29 14:28 .
drwxr-xr-x. 4 tangmuna hubmap      4096 Mar 28 16:07 ..
-rw-r--r--. 1 tangmuna hubmap    625910 Mar 27 16:12 dispersion_plot.pdf
-rw-r--r--. 1 tangmuna hubmap 202017752 Mar 27 16:12 expr.h5ad
drwxr-xr-x. 3 tangmuna hubmap      4096 Mar 27 15:43 fastqc_output           <-- directory
-rw-rw-r--. 1 tangmuna hubmap       106 Mar 27 16:12 genome_build.json
drwxr-xr-x. 4 tangmuna hubmap      4096 Mar 28 16:07 hubmap-ui               <-- directory
-rw-r--r--. 1 tangmuna hubmap     24095 Mar 27 16:12 marker_genes_by_cluster_logreg.pdf
-rw-r--r--. 1 tangmuna hubmap     24094 Mar 27 16:12 marker_genes_by_cluster_t_test.pdf
-rw-r--r--. 1 tangmuna hubmap   3357736 Mar 27 16:13 qc_results.hdf5
-rw-r--r--. 1 tangmuna hubmap       455 Mar 27 16:13 qc_results.json
-rw-r--r--. 1 tangmuna hubmap  91487384 Mar 27 16:12 raw_expr.h5ad
drwxr-xr-x. 6 tangmuna hubmap      4096 Mar 27 16:11 salmon_out              <-- directory
-rw-r--r--. 1 tangmuna hubmap 440616999 Mar 27 16:16 scvelo_annotated.h5ad
-rw-r--r--. 1 tangmuna hubmap    199081 Mar 27 16:16 scvelo_embedding_grid.pdf
-rw-r--r--. 1 tangmuna hubmap 558008647 Mar 27 16:45 secondary_analysis.h5ad
-rw-r--r--. 1 tangmuna hubmap     83837 Mar 27 16:12 umap_by_leiden_cluster.pdf
-rw-r--r--. 1 tangmuna hubmap    690387 Mar 27 16:12 umap_embedding_density.pdf


- expr.h5ad: Raw gene expression, in HDF5 format, readable with the 'anndata' Python package, intronic counts as AnnData layer (Format: EDAM_1.24.format_3590)

- raw_expr.h5ad: Raw gene expression, in HDF5 format, readable with the 'anndata' Python package, intronic counts as AnnData layer (Format: EDAM_1.24.format_3590)

- scvelo_annotated.h5ad: RNA velocity results, in HDF5 format, readable with the 'anndata' Python package (Format: EDAM_1.24.format_3590)

- secondary_analysis.h5ad: Normalized gene expression with additional metadata, in HDF5 format, readable with the 'anndata' Python package (Format: EDAM_1.24.format_3590)

- genome_build.json: genome build information (Format: EDAM_1.24.format_3464)

- dispersion_plot.pdf: gene expression (Format: EDAM_1.24.format_3508)
- qc_results.hdf5: quality control results from Scanpy, per-cell and per-gene (Format: EDAM_1.24.format_3590)
- qc_results.json: quality control report (Format: EDAM_1.24.format_3464)
- scvelo_embedding_grid.pdf: RNA velocity grid plot (Format: EDAM_1.24.format_3508)
- marker_genes_by_cluster_logreg.pdf
- marker_genes_by_cluster_t_test.pdf
- umap_by_leiden_cluster.pdf : UMAP plot of cells, colored by Leiden cluster ID (Format: EDAM_1.24.format_3508)
- umap_embedding_density.pdf : UMAP plot of cells, colored by embedding density (Format: EDAM_1.24.format_3508)

- fastqc_output/raw/fastq/RNA/: QA report per seq file in html format (Format: EDAM_1.24.format_2331)
  - *.html
> [tangmuna@vm010 outputs]$ ls fastqc_output/raw/fastq/RNA/
1_1_012618_H17w_Co_2_1_S5.R1_fastqc.html  1_1_012618_H17w_Co_2-1_S5.R2_fastqc.zip   1_1_012618_H17w_Co_2-2_S6.R2_fastqc.html  1_1_012618_H17w_Co_2-3_S7.R1_fastqc.zip
1_1_012618_H17w_Co_2-4_S8.R1_fastqc.html  1_1_012618_H17w_Co_2-4_S8.R2_fastqc.zip   1_1_012618_H17w_Co_2_1_S5.R1_fastqc.zip   1_1_012618_H17w_Co_2-2_S6.R1_fastqc.html
1_1_012618_H17w_Co_2-2_S6.R2_fastqc.zip   1_1_012618_H17w_Co_2-3_S7.R2_fastqc.html  1_1_012618_H17w_Co_2-4_S8.R1_fastqc.zip   1_1_012618_H17w_Co_2-1_S5.R2_fastqc.html
1_1_012618_H17w_Co_2-2_S6.R1_fastqc.zip   1_1_012618_H17w_Co_2-3_S7.R1_fastqc.html  1_1_012618_H17w_Co_2-3_S7.R2_fastqc.zip   1_1_012618_H17w_Co_2-4_S8.R2_fastqc.html

- hubmap_ui
  - anndata-zarr
    - scvelo_annotated.zarr
      - .zgroup : AnnData Zarr store for storing and visualizing single cell sequencing outputs of velocity analysis. (Format: EDAM_1.24.format_2333)
      - layers
        - .zgroup
	- spliced
	  - .zgroup
	- spliced_unspliced_sum
 	  - .zgroup
	- unspliced
	  - .zgroup	
      - obs
        - .zgroup
	- _categories
	  - .zgroup
      - obsm
        - .zgroup
      - obsp
        - .zgroup      
	- connectivities
	  - .zgroup	
	- distances
	  - .zgroup	
      - uns
        - .zgroup
        - leiden
	  - .zgroup
	  - params
  	    - .zgroup		  
	- neighbors
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- pca
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- recover_dynamics
	  - .zgroup		
	- umap
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- velocity_graph
	  - .zgroup
        - velocity_graph_neg
	  - .zgroup	
	- velocity_params
	  - .zgroup	
      - var
        - .zgroup
	- _categories
	  - .zgroup
      - varm
        - .zgroup
    - secondary_analysis.zarr (under hubmap_ui/anndata-zarr)
      - .zgroup : AnnData Zarr store for storing and visualizing single cell sequencing outputs of velocity analysis. (Format: EDAM_1.24.format_2333)
      - layers
        - .zgroup
	- spliced
	  - .zgroup
	- spliced_unspliced_sum
 	  - .zgroup
	- unscaled
	  - .zgroup	
	- unspliced
	  - .zgroup	
      - obs
        - .zgroup
	- _categories
	  - .zgroup
      - obsm
        - .zgroup
      - obsp
        - .zgroup      
	- connectivities
	  - .zgroup	
	- distances
	  - .zgroup	
      - uns
        - .zgroup
        - annotation_metadata
	  - .zgroup
	- hvg
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- leiden
	  - .zgroup	
	  - params
  	    - .zgroup
	- neighbors
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- pca
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- rank_gene_groups
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- umap
	  - .zgroup	
	  - params
  	    - .zgroup		  
	- umap_density_params
	  - .zgroup	
      - var
        - .zgroup
	- _categories
	  - .zgroup
      - varm
        - .zgroup            
  - output (under hubmap_ui)
    - secondary_analysis.arrow : Input data relevant for visualization saved in columnar Apache Arrow format. (Format: EDAM_1.24.format_2333)
    - secondary_analysis.cell-sets.json : JSON-formatted information about the hierarchy scRNA-seq's cells. (Format: EDAM_1.24.format_3464)
    - secondary_analysis.cells.json :  JSON-formatted information about this scRNA-seq run including scatterplot coordinates and clustering. (Format: EDAM_1.24.format_3464)"
    - secondary_analysis.csv : Input data relevant for visualization saved in columnar comma-separated-file format. (Format: EDAM_1.24.format_3752)
    - secondary_analysis.factors.json : JSON-formatted information about this scRNA-seq's clustering. (Format: EDAM_1.24.format_3464)

- salmon_out
  - alevin
      alevin.log       quants_mat_cols.txt  quants_mat.mtx.gz    quants_tier_mat.gz
      featureDump.txt  quants_mat.gz        quants_mat_rows.txt  whitelist.txt
  - aux_info
      alevin_meta_info.json  ambig_info.tsv  expected_bias.gz  fld.gz  meta_info.json  observed_bias_3p.gz  observed_bias.gz
  - cmd_info.json
  - lib_format_counts.json
  - libParams
    - flenDist.txt
  - logs
    - salmon_quant.log
    
Note: https://portal.hubmapconsortium.org/browse/dataset/f47349f1e0fea69263a8d68c7dac65c5#provenance to see explanation of those files